include/ndarray/pybind11.h

```diff
@@ -57,188 +175,223 @@ NAMESPACE_BEGIN(detail)
  */
 template <typename T, int N, int C>
 class type_caster< ndarray::Array<T,N,C> > {
+    using Helper = ndarray::Pybind11Helper<T,N,C>;
 public:
+
     bool load(handle src, bool) {
-        _none = src.is_none();
-        if (_none) return true;
-        _src.reset(src.ptr()); // equivalent to add_ref=true, keep alive for stage 2
-        if (!ndarray::PyConverter< ndarray::Array<T,N,C> >::fromPythonStage1(_src)) {
-            PyErr_Clear();
-            return false;
-        }
-        return true;
-    }
-    static handle cast(const ndarray::Array<T,N,C> &src, return_value_policy /* policy */, handle /* parent */) {
-        return ndarray::PyConverter< ndarray::Array<T,N,C> >::toPython(src);
+        return _helper.init(src) && _helper.check();
```

Comment:
Should we do it this way? I'd rather see a check of the input before initialization / blowing away the previously held value. I'm also not that convinced that `_helper` needs quite as much (if any) state. But I could be missing something.

Reply:
I don't actually know how pybind11 will call these member functions - does it ever actually call `load` repeatedly on the same object?  In any case, if it does, I wouldn't want to keep the old value unless we could somehow know it was appropriate for the new arguments.

In case this is the source of our differences, note that the diff here is misleading - the line above is the first line of `load`, not the first line of `cast`.